StRoot/StFstRawHitMaker/StFstRawHitMaker.cxx

```diff
@@ -390,6 +391,60 @@ Int_t StFstRawHitMaker::Make()
 
     LOG_INFO << " Total number of FST Raw Hits - Step II = " << mFstCollectionPtr->getNumRawHits() << " w/ idTruth = " << nIdTruth_Fst << endm;
 
+    // Fill Fst raw hits we got so far into StEvent if the chain option is called
+    if(IAttr("fstEvtRawHit")) {//True for storing FST Raw hits
+        LOG_INFO << " Filling FST Raw Hits into StEvent" << endm;
+
+        //prepare for StEvent
+        mEvent = (StEvent *) GetInputDS("StEvent");
+
+        if(mEvent) {
+            LOG_DEBUG<<"Found StEvent"<<endm;
+        } else {
+            mEvent = new StEvent();
+            AddData(mEvent);
+            LOG_DEBUG <<"Added StEvent"<<endm;
+        }
+
+        // Get pointer to an existing StFstEvtCollection if any
+        mFstEvtCollection = mEvent->fstEvtCollection();
+
+        // If no fst raw hit collection, create one
+        if (!mFstEvtCollection) {
+            mFstEvtCollection = new StFstEvtCollection();
+            mEvent->setFstEvtCollection(mFstEvtCollection);
+            LOG_DEBUG << "Make() - Added new StFstEvtCollection to this StEvent" << endm;
+        }
+
+        if(mFstCollectionPtr->getNumRawHits() > 0) {
+            for(int wedgeIdx=0; wedgeIdx<kFstNumWedges; ++wedgeIdx ){
+                StFstRawHitCollection *rawHitCollectionPtr = mFstCollectionPtr->getRawHitCollection( wedgeIdx );
+                if( rawHitCollectionPtr ){
+                    std::vector<StFstRawHit*>& rawHitVec = rawHitCollectionPtr->getRawHitVec();
+                    std::vector< StFstRawHit* >::iterator rawHitIter;
+
+                    for( rawHitIter = rawHitVec.begin(); rawHitIter != rawHitVec.end(); ++rawHitIter ){
+                        StFstRawHit* rawHit = *rawHitIter;
+
+                        StFstRawHit* newRawHit = new StFstRawHit();
+                        newRawHit->setChannelId( rawHit->getChannelId() );
+                        newRawHit->setGeoId( rawHit->getGeoId() );
+                        newRawHit->setSeedhitflag( rawHit->getSeedhitflag() );
+                        for(int itb=0; itb<kFstNumTimeBins; itb++){
+                            newRawHit->setCharge( rawHit->getCharge(itb), itb );
+                            newRawHit->setChargeErr( rawHit->getChargeErr(itb), itb );
+                        }
+                        newRawHit->setMaxTimeBin( (int)rawHit->getMaxTimeBin() );
+                        newRawHit->setDefaultTimeBin( (int)rawHit->getDefaultTimeBin() );
+                        newRawHit->setIdTruth( rawHit->getIdTruth() );
```

Comment:
Please write a copy constructor for StFstRawHit and use it here in place of these lines

Reply:
Done.